packages/edit-post/src/components/options-modal/options/enable-custom-fields.js

```diff
@@ -1,47 +1,54 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
 import { withSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import BaseOption from './base';
 
-export class EnableCustomFieldsOption extends Component {
-	constructor( { isChecked } ) {
-		super( ...arguments );
-
-		this.toggleCustomFields = this.toggleCustomFields.bind( this );
-
-		this.state = { isChecked };
-	}
-
-	toggleCustomFields() {
-		// Submit a hidden form which triggers the toggle_custom_fields admin action.
-		// This action will toggle the setting and reload the editor with the meta box
-		// assets included on the page.
-		document.getElementById( 'toggle-custom-fields-form' ).submit();
-
-		// Make it look like something happened while the page reloads.
-		this.setState( { isChecked: ! this.props.isChecked } );
-	}
-
-	render() {
-		const { label } = this.props;
-		const { isChecked } = this.state;
+export function CustomFieldsConfirmation() {
+	const [ isReloading, setIsReloading ] = useState( false );
+
+	return (
+		<>
+			<p className="edit-post-options-modal__custom-fields-confirmation-message">
+				{ __( 'Page reload is required for this change.' ) }
+			</p>
+			<Button
+				className="edit-post-options-modal__custom-fields-confirmation-button"
+				isDefault
+				isBusy={ isReloading }
+				disabled={ isReloading }
+				onClick={ () => {
+					setIsReloading( true );
+					document.getElementById( 'toggle-custom-fields-form' ).submit();
+				} }
+			>
+				{ __( 'Save & Reload' ) }
```

Comment:
'Save & Reload' - clicking the button does save the change to the custom fields option, but doesn't save any other edits to the post, which may be misleading to the user and lead to lost data. I'm thinking that perhaps the post should be saved when this is clicked?

Reply:
In case there is unsaved content, the page won't reload automatically and will still show the "Are you sure" dialog of the browser. While this is not ideal, I think that merging this will still improve the totally unclear and inaccessible way that is shipped now. I think that mixing together post saving could become very complex (you need to wait for the request to succeed etc) and we would need to handle possible edge cases. 

Do you think that updating the label to remove the word "Save" would solve the problem in a way we can get this merged and fix the root accessibility problem? I'm thinking of making the label dynamic based on the checkbox status to reflect the exact action that will be taken — either "Enable & Reload" or "Disable & Reload".

Reply:
And/or add something in lines of "Make sure your editor content is saved before proceeding"